File: jar-common/src/main/java/com/beancounter/common/input/TrustedTrnImportRequest.java
package com.beancounter.common.input;

import com.beancounter.common.model.CallerRef;
import com.beancounter.common.model.Portfolio;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TrustedTrnImportRequest implements TrnImport {
  private Portfolio portfolio;
  private CallerRef callerRef;
  private String message;
  private List<String> row;
}

File: jar-common/src/main/kotlin/com/beancounter/common/input/TrustedTrnImportRequest.kt
package com.beancounter.common.input

import com.beancounter.common.model.CallerRef
import com.beancounter.common.model.Portfolio

/**
 * Contract to write a transaction, transforming from an ImportFormat, into BC for the requested Portfolio.
 * Ownership of the portfolio is not verified, making this a trusted request.
 *
 * Message is optional and will be output to the log when the event is processed
 */
data class TrustedTrnImportRequest(
    override var portfolio: Portfolio,
    override val importFormat: ImportFormat = ImportFormat.BC,
    override var message: String = "",
    val row: List<String> = emptyList(),
    var callerRef: CallerRef = CallerRef(portfolio.owner.id, row[1], row[2]),
) : TrnImport
